assets/js/translator.js

```diff
@@ -699,9 +706,13 @@ function translateText(ignoreIfEmpty) {
 function translateDoc() {
     var validPair = pairs[curSrcLang] && pairs[curSrcLang].indexOf(curDstLang) !== -1;
 
-    var file = droppedFile ? droppedFile : undefined;
-    if($('input#fileInput')[0].files.length > 0 && $('input#fileInput')[0].files[0].length !== 0) {
-        file = $('input#fileInput')[0].files[0];
+    var file = droppedFile ? droppedFile : null;
+
+    // Flow can only infer type HTMLElement
+    var fileInput /*: HTMLInputElement */ = ($('input#fileInput')[0]/*: any */);
```

Comment:
Can this not read:

```
var fileInput = ($('input#fileInput')[0] /*: HTMLInputElement */);
```

?

Reply:
Thought so, too. Apparently not. Because this isn't guaranteed to succeed, Flow doesn't permit.

Reply:
Couldn't find a more "idiomatic" way to do it. This is kinda ugly.

Reply:
That's alright - not a big deal.